edx_proctoring/api.py

```diff
@@ -1526,6 +1526,23 @@ def _calculate_allowed_mins(due_datetime, allowed_mins):
     return actual_allowed_mins, is_exam_past_due_date
 
 
+def _is_self_paced_course(user_id, course_id):
```

Comment:
This introduces the concept of self-paced course into proctoring module. The fix here is quite intrusive to me.
When we call the function `_get_timed_exam_view`, we did pass in a `context` object where contains the `due_date` from LMS. My question is why can't we leverage this context object instead?